providers/gitlab.go

```diff
@@ -250,6 +295,17 @@ func (p *GitLabProvider) GetEmailAddress(ctx context.Context, s *sessions.Sessio
 		return "", fmt.Errorf("group membership check failed: %v", err)
 	}
 
+	// Check project membership
```

Comment:
Can we make this a `verifyProjectMembership` function to mirror the organization of the other verifiers that are similar to this above?